pangeo-notebook/conda-lock.yml

```diff
@@ -1113,77 +1114,80 @@ package:
     scipy: '>=1.5'
     setuptools: '>=42'
     toolz: '>=0.8.2'
-    xarray: '>=0.18'
-  url: https://conda.anaconda.org/conda-forge/noarch/argopy-0.1.14-pyhd8ed1ab_0.conda
+    xarray: '>=0.18,<2024.3'
+  url: https://conda.anaconda.org/conda-forge/noarch/argopy-0.1.16-pyhd8ed1ab_0.conda
```

Comment:
> not sure why this downgrade happened...

Tracked it down to argopy pinning to `xarray<2024.3` as mentioned in their changelog https://github.com/euroargodev/argopy/blob/2f2e2dc884cf5d3b39aa38339e1027260fc4c60d/docs/whats-new.rst?plain=1#L57. Seems to be something about https://github.com/pydata/xarray/issues/8909

Reply:
Hmmm, my brief understanding is there is an issue specifically for the case of writing out netCDF files with the `scipy` engine. 

I prefer to favor supporting the most general packages like xarray when conflicts with more domain-specific packages (like argopy) come up. One positive outcome of these images is just motivating broad ecosystem compatibility and fixing certain upstream issues eventually! But short-term I think pinning the domain-specific package is best to not block other updates like this.